File: app/javascript/channels/messages_channel.js
import consumer from "./consumer"

console.log("messages_channel.js loaded");

  consumer.subscriptions.create("MessagesChannel", {
  connected() {
    // Called when the subscription is ready for use on the server
    const roomId = $("#chat-box").data("room-id");
    this.checkIn(roomId);
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
  },

  received(data) {
    // Called when there's incoming data on the websocket for this channel
    
    $("#chat-box").append(data);
    $("#message-field").val('');

    const posts = $(".message-row").length;
    if (posts > 10) {
      $(".message-row").slice(10).remove();
      // $("message-row).first().remove();
    }
  },

  checkIn(roomId) {
    if (roomId) {
      this.perform('checkIn', { room_id: roomId });
    } else {
      this.perform('checkOut');
    }
  }
});

File: app/channels/messages_channel.rb
class MessagesChannel < ApplicationCable::Channel
  def subscribed
    # stream_from "messages_channel"
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  def checkIn(data)
    # ActionCable.server.broadcast("messages_channel", message: data['message'])
    # Message.create!(content: data['message'])
    room_id = "#{data['room_id']}"
    stream_from "messages_room_#{room_id}"
  end

  def checkOut
    stop_all_streams
  end
end
